src/statistics.jl

```diff
@@ -0,0 +1,46 @@
+abstract type AbstractStatistic end
+
+struct Mean{D<:AbstractDomain} <: AbstractStatistic
+   domain::D
+   values::Dict{Symbol, Vector} 
+end
+
+Mean(domain, values) = Mean{typeof(domain)}(domain, values)
+
+function Mean(solution::SimulationSolution)
+    # solution variables
+    variables = collect(keys(solution.realizations))
+
+    # output dictionary
+    values = Dict(var => mean(solution.realizations[var]) for var in variables)
+    
+    Mean(solution.domain, values)
```

Comment:
Here you could call the internal constructor `Mean{typeof(sdomain)}(sdomain, values)`